Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -626,29 +641,22 @@ void CGPathApply(CGPathRef path, void* info, CGPathApplierFunction function) {
 }
 
 /**
- @Status Interoperable
- @Notes
+ @Status Caveat
+ @Notes eoFill ignored. Default fill pattern for ID2D1 Geometry is used.
 */
 bool CGPathContainsPoint(CGPathRef path, const CGAffineTransform* transform, CGPoint point, bool eoFill) {
+    RETURN_FALSE_IF(!path);
+
     if (transform) {
         point = CGPointApplyAffineTransform(point, *transform);
     }
-    // check if the point is outside this box already, if it is, return false
-    CGRect boundingBox = CGPathGetBoundingBox(path);
-    if (!CGRectContainsPoint(boundingBox, point)) {
-        return false;
-    }
 
-    CGContextRef context =
-        CGBitmapContextCreate(0, boundingBox.origin.x + boundingBox.size.width, boundingBox.origin.y + boundingBox.size.height, 1, 1, 0, 0);
+    BOOL containsPoint = FALSE;
 
-    CGContextAddPath(context, path);
-
-    bool inPath = CGContextIsPointInPath(context, eoFill, point.x, point.y);
-
-    CGContextRelease(context);
+    path->closePath();
+    FAIL_FAST_IF_FAILED(path->_impl.pathGeometry->FillContainsPoint(_CGPointToD2D_F(point), D2D1::IdentityMatrix(), &containsPoint));
```

Comment:
Talked to @DHowett-MSFT on how to standardize this. *All* internal method that have some failure state should be unified to do the same thing (likely HRESULT return as they call windows-y things). At the outermost layer, the determination is made for that function what style it should be. CG appears to be a generally "no fail" set of apis so likely this means FAIL_FAST for us until we determine to dial it back. If we dial it back we should LOG and early return in any failure cases to preserve the code flow and keep it clean. Sound good?

---
In reply to: [85607025](https://github.com/Microsoft/WinObjC/pull/1255#discussion_r85607025) [](ancestors = 85607025,85594909)